src/remote-config/remote-config-api-client.ts

```diff
@@ -160,6 +160,43 @@ export class RemoteConfigApiClient {
       });
   }
 
+  public publishTemplate(template: RemoteConfigTemplate, options?: { force: boolean }): Promise<RemoteConfigTemplate> {
+    let ifMatch: string = template.etag;
+    if (options && options.force == true) {
+      // setting `If-Match: *` forces the Remote Config template to be updated
+      // and circumvent the ETag, and the protection from that it provides.
+      ifMatch = '*';
+    }
+    return this.getUrl()
+      .then((url) => {
+        const request: HttpRequestConfig = {
```

Comment:
This block is pretty similar to what we use in `validateTemplate`. I considered moving this to a separate function and make the code reusable, but it seemed like an overkill for this. I am open to suggestions though.

Reply:
May be a private `publishTemplate(template, etag)` method makes sense.